src/server/internalServer.cpp

```diff
@@ -775,7 +778,13 @@ std::unique_ptr<Response> InternalServer::handle_content(const RequestContext& r
 
   const std::shared_ptr<Reader> reader = get_reader(bookName);
   if (reader == nullptr) {
-    return Response::build_404(*this, request, bookName);
+    std::string searchURL = m_root+"/search?pattern="+pattern; // Make a full search on the entire library.
+    MustacheData data;
+    data.set("pattern", kiwix::urlDecode(pattern));
+    data.set("searchURL", searchURL);
+    std::string details = tmpl.render(data);
```

Comment:
Convert this piece of code to a small helper function, so that it is replaced with a single line:

```c++
const std::string details = searchSuggestionHTML(searchURL, kiwix::urlDecode(pattern));
```

Declare/define the `searchSuggestionHTML()` function in an unnamed namespace. If you can think of a better name for it, go ahead and do it.